pallets/jobs/src/lib.rs

```diff
@@ -258,23 +264,22 @@ pub mod module {
 
 			let job_id = Self::get_next_job_id()?;
 			let role_type = job.job_type.get_role_type();
+			let participants =
+				job.job_type.clone().get_participants().ok_or(Error::<T>::InvalidJobPhase)?;
```

Comment:
This will error if the submitted job is not phase one.

Reply:
A better option, since you seem to need this value down there to emit the event, you create a mutable vec at the top here, and assign its value in each of the if branches.